src/usb.c

```diff
@@ -189,7 +189,7 @@ int _stlink_usb_version(stlink_t *sl) {
         cmd[i++] = STLINK_APIV3_GET_VERSION_EX;
 
         size = send_recv(slu, 1, cmd, slu->cmd_len, data, rep_len);
-        if (size != rep_len) {
+        if (size == -1 || (uint32_t)(size) != rep_len) {
```

Comment:
Please check if it works with `size != (ssize_t)rep_len` which is the current state in the develop branch.

Reply:
It does work with the develop branch. Sorry I should have read the README.md file more closely.

Thanks for your review.

Closing this PR

Reply:
No problem. Thx for your attentiveness and addressing this issue though.